eth/db/witness.py

```diff
@@ -0,0 +1,61 @@
+from typing import (
+    Collection,
+    Dict,
+    Set,
+    Tuple,
+)
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+
+from eth.abc import (
+    WitnessIndexAPI,
+)
+
+
+class WitnessIndex(WitnessIndexAPI):
+    def __init__(
+            self,
+            witness_hashes: Set[Hash32],
+            accounts_metadata_queried: Dict[Address, Tuple[bool, Tuple[int, ...]]]) -> None:
+
+        self._trie_node_hashes = tuple(witness_hashes)
+        self._accounts_metadata_queried = accounts_metadata_queried
+
+    @property
+    def hashes(self) -> Tuple[Hash32, ...]:
+        return self._trie_node_hashes
+
+    @property
+    def accounts_queried(self) -> Collection[Address]:
+        return self._accounts_metadata_queried.keys()
+
+    @property
+    def account_bytecodes_queried(self) -> Tuple[Address, ...]:
+        return tuple(
+            address
+            for address, (was_bytecode_queried, _slots) in self._accounts_metadata_queried.items()
```

Comment:
Named tuple approach would mean you wouldn't have to unpack the second value and you could just access the fields using properties.